lib/msal-angular/src/msal.service.ts

```diff
@@ -66,324 +61,151 @@ export class MsalService extends UserAgentApplication {
-                    //redirect to redirect uri. No page reload here since we are only removing the url after the hash
-                    else {
-                        window.location.hash = '';
-                    }
-                }
-            }
-            else {
-                // state did not match, broadcast an error
-                this.broadcastService.broadcast("msal:stateMismatch", msalError);
-            }
-        }
-        else {
-            var pendingCallback = this._cacheStorage.getItem(Constants.urlHash);
-            if (pendingCallback) {
-               this.processRedirectCallBack(pendingCallback);
-            }
-        }
-    };
-
-
-    private processRedirectCallBack(hash: string): void {
-        this._logger.info('Processing the callback from redirect response');
-        const requestInfo = this.getRequestInfo(hash);
-        const token = requestInfo.parameters[Constants.accessToken] || requestInfo.parameters[Constants.idToken];
-        const errorDesc = requestInfo.parameters[Constants.errorDescription];
-        const error = requestInfo.parameters[Constants.error];
-        var tokenType: string;
-        this._cacheStorage.removeItem(Constants.urlHash);
-        var msalError = new MSALError(error, errorDesc);
-        var authenticationResult = new AuthenticationResult(token);
-        if (requestInfo.parameters[Constants.accessToken]) {
-            tokenType = Constants.accessToken;
-            if (token) {
-                authenticationResult.tokenType= tokenType;
-                this.broadcastService.broadcast("msal:acquireTokenSuccess", authenticationResult);
-            }
-            else if (error && errorDesc) {
-                //TODO this should also send back the scopes
-                this.broadcastService.broadcast("msal:acquireTokenFailure", msalError);
-            }
-        }
-        else {
-            tokenType = Constants.idToken;
-            if (token) {
-                authenticationResult.tokenType= tokenType;
-                this.broadcastService.broadcast("msal:loginSuccess", authenticationResult);
-            }
-            else if (error && errorDesc) {
-                this.broadcastService.broadcast("msal:loginFailure", msalError);
-            }
-        }
+        });
     }
 
+    private isUnprotectedResource(url: string): boolean {
+        const unprotectedResources = (this.msalConfig.framework && this.msalConfig.framework.unprotectedResources) || this.msalAngularConfig.unprotectedResources;
 
-    private isUnprotectedResource(url: string) {
-        if (this.config && this.config.unprotectedResources) {
-            for (var i = 0; i < this.config.unprotectedResources.length; i++) {
-                if (url.indexOf(this.config.unprotectedResources[i]) > -1) {
-                    return true;
-                }
-            }
-        }
-        return false;
+        return unprotectedResources.some(resource => url.indexOf(resource) > -1);
```

Comment:
I hope you won't mind if I add a comment to your PR as a simple beta tester 😉
If you want that I open an issue instead, please let me know!

I have two issues with this single line.

- I had no unprotected resources in my Angular application, and I first didn't specify it in the `MsalAngularConfiguration` object given to configure the module. So it was undefined, and throws an error here ("Cannot call method some of undefined"). A simple workaround is to explicitly specify it as an empty array, but the code could be more protective, or the object given as the second argument of `MsalModule.forRoot` could be merged with the default one.

-  The second issue is more problematic, and concerns the URL matching. I had collisions between protected API URLs, and Angular route's paths, which are now automatically pushed in the `unprotectedResources` array. For example, I have protected calls to an API hosted at "https://api.com/contracts" and an Angular route "contracts".
As the matching is done with a "contains" (`url.indexOf(resource) > -1`), the interceptor considered my API calls as unprotected, and didn't include a token for it. Wouldn't it be safer to match URLs with a `startsWith`?

Reply:
@asiffermann Of course, appreciate the feedback!

1. `unprotectedResources`: This will be fixed by #1291 
2. URL matching: Yeah, this seems reasonable. I'll investigate and follow up.

Reply:
Fix for unprotectedResources has been included in this PR, interceptor will be fixed incrementally.